UM/Settings/AdditionalSettingDefinitionAppender.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2023 UltiMaker.
+# Uranium is released under the terms of the LGPLv3 or higher.
+
+import json
+from pathlib import Path
+import os.path
+from typing import Any, Dict, List
+
+from UM.Logger import Logger
+from UM.PluginObject import PluginObject
+from UM.Settings.SettingDefinition import DefinitionPropertyType, SettingDefinition
+
+
+def prependIdToSettings(tag_type: str, tag_id: str, tag_version: str, settings: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Consider making this a static function of the class. My main argument for it is that right now, the documentation (that is nice and extensive!) for this function takes away the attention of the documentation of the class. But the class should be the "meat" of what is going on (and thus, should probably be listed first)